### PR TITLE
[expo-status-bar] Simple new module that wraps React Native StatusBar

### LINKED ISSUE #8002: Status bar white when starting app on iOS with dark mode enabled even if light mode is forced
The app in this video is an empty new app that has no `userInterfaceStyle` config and therefore is locked to light mode.

https://streamable.com/vlzv6e

**Expected behaviour**: status bar content default to dark when OS dark mode is enabled but app is locked to light
**Actual behaviour**: status bar content defaults to light

This occurs because `UIViewControllerBasedStatusBarAppearance` is false by default in React Native and Expo apps in order to support the react-native `StatusBar` component. With this configuration set, `UIStatusBarStyleDefault` is white when dark mode is enabled on the OS and dark otherwise, even if we set the user interface style on the presented view controller.

### LINKED ISSUE #8436: Status bar on Android switches to not-translucent after mounting StatusBar component
### Environment

Expo CLI 3.20.9 environment info:
System:
OS: macOS 10.15.4
Shell: 3.2.57 - /bin/bash
Binaries:
Node: 10.17.0 - ~/.nvm/versions/node/v10.17.0/bin/node
Yarn: 1.22.0 - /usr/local/bin/yarn
npm: 6.11.3 - ~/.nvm/versions/node/v10.17.0/bin/npm
Watchman: 4.9.0 - /usr/local/bin/watchman
IDEs:
Android Studio: 3.5 AI-191.8026.42.35.6010548
Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
npmPackages:
expo: ^37.0.10 => 37.0.10
react: ~16.9.0 => 16.9.0
react-dom: ~16.9.0 => 16.9.0
react-native: ~0.61.5 => 0.61.5
react-native-web: ~0.11.7 => 0.11.7
react-navigation: 4.0.10 => 4.0.10
npmGlobalPackages:
expo-cli: 3.20.9

### Steps to Reproduce

1. Create an app with the default `app.json` settings, which include a translucent Android status bar
2. Add `<StatusBar />` or `<StatusBar translucent />` (or really any `StatusBar` component other than one that explicitly sets `translucent` to false) to render.
3. Run on Android.
4. The status bar will not be translucent as soon as the `StatusBar` is mounted, even if `translucent` is true.

### Expected Behavior

As long as `translucent` is not set to false or `backgroundColor` is not set on the `StatusBar` component, mounting a `StatusBar` component should not remove the default translucency. This is how it behaved on Expo 36 with the Expo 36 client and Expo 36 standalone apps.

### Actual Behavior

Mounting a `StatusBar` component removes the translucency and also shifts the entire view down to accommodate the non-translucent status bar.

This happens on Expo 37 with both the Expo 37 client and standalone apps.
This also happens when running an Expo 36 client on the Expo 37 client, but does not happen in the Expo 36 standalone app.

### Reproducible Demo

Snack demonstrating issue: https://snack.expo.io/@llamaluvr/expo-37-statusbar-issue

This shows a common scenario- hiding a `StatusBar` and then showing it again. Usually this is done in conjunction with react-navigation (e.g., hide `StatusBar` on one screen, show on another). I left that out, but the effect is the same with navigation.

Press "Hide" and then "Show" and the status bar will no longer be translucent.
Also, you can try just pressing "Show" and the same thing happens. Just the act of mounting the `StatusBar` component causes the issue.

Here's a control experiment using the imperative API: https://snack.expo.io/@llamaluvr/201cf5
The imperative API does not have the same issue, though I'm not sure if it's practical to refactor all component uses of `StatusBar` to the imperative API.

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

First pass at a simple wrapper around StatusBar to make it work more nicely for Expo users. Things like automatically making it non-translucent on Android and "default" not working right on iOS with color schemes are not pleasant for users. Both of these came up with issues during dogfooding.

Fixes #8436 
Fixes #8002

**Bonus!** this PR also converts the tabs template to support dark mode. This was because we didn't have a great way to test expo-status-bar behavior in any existing apps inside of this repository, and we wanted to add support for dark mode in tabs soon anyways.

# How

Init a new module, remove native code, add some little wrapper tsx files.

# TODO

- [x] Write docs page
- [x] Verify with others that this is reasonable
- [x] Expand test coverage
- [x] Manually test in a variety of common cases

# Test Plan

- Run unit tests
- Try it out in tabs template

### CODE REVIEW

File: packages/expo-status-bar/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "directory": "packages/expo-status-bar"
+  },
+  "bugs": {
+    "url": "https://github.com/expo/expo/issues"
+  },
+  "author": "650 Industries, Inc.",
+  "license": "MIT",
+  "homepage": "https://docs.expo.io/versions/latest/sdk/module-template",
+  "dependencies": {
+    "expo-module-scripts": "^1.2.0"
```
Comment (tsapeta) [inline packages/expo-status-bar/package.json:33]: Shouldn't it be a dev dependency?

File: packages/expo-status-bar/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (tsapeta) [inline packages/expo-status-bar/tsconfig.json:1]: We used to be using `tsconfig.json` generated from `expo-module-scripts` 🤔

File: packages/expo-status-bar/babel.config.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module.exports = function(api) {
```
Comment (tsapeta) [inline packages/expo-status-bar/babel.config.js:1]: Same here, it should be generated by `expo-module-scripts`

File: packages/expo-status-bar/package.json (hunk 1 above)
Comment (brentvatne) [reply packages/expo-status-bar/package.json:33]: yeah.. hmm this was just created by `et cu --name expo-status-bar` -- i guess the template is outdated

File: packages/expo-status-bar/tsconfig.json (hunk 2 above)
Comment (brentvatne) [reply packages/expo-status-bar/tsconfig.json:1]: hm, another `et cu` thing, will look into it

**Comment by brentvatne** [comment]
Also, this fixes https://github.com/expo/expo/issues/8436

**Comment by brentvatne** [comment]
Fixes https://github.com/expo/expo/issues/8002

**Review by tsapeta — APPROVED** [review]

File: docs/pages/guides/configuring-statusbar.md [hunk 4]
```diff
@@ -36,19 +54,64 @@ Defaults to `true`.
...
+      <StatusBar style="light" />
+    </View>
+  );
+}
+```
+
+<details><summary><h4>How is expo-status-bar different from the StatusBar component included in React Native?</h4></summary>
+<p>
+
+`expo-status-bar` builds on top of the `StatusBar` component that React Native provides in order to give you better defaults when you're building an app with Expo tools. For example, the `translucent` property of `expo-status-bar` defaults to `true` or, if you have changed that property in `androidStatusBar`, it will use that value instead. The default in React Native for `translucent` is always `false`, which can be confusing when in projects created using Expo tools, because the default is `true` for consistency with iOS.
```
Comment (ide) [inline docs/pages/guides/configuring-statusbar.md:94]: Remove "tools" in this paragraph, just "Expo".

File: docs/pages/guides/configuring-statusbar.md [hunk 5]
```diff
@@ -36,19 +54,64 @@ Defaults to `true`.
...
 
-If you need to use such a library, there are a few options:
+Libraries like [React Navigation](../../guides/routing-and-navigation/) will handle this for you when the UI that they provide overlap with the status bar. You are likely to encounter cases where you will need to manually adjust your layout to prevent some content (such as text) from being rendered underneath it. To do this, we recommend using [react-native-safe-area-context](/versions/latest/sdk/safe-area-context/) to find the safe area insets and add padding or margins to your layout accordingly.
+
+## Working with misbehaving 3rd-party Libraries
+
+Projects initialized with Expo tools make the status bar `translucent` by default on Android. This is consistent with iOS and more in line with material design. Unfortunately, some libraries don't support `translucent` status bars. This is generally bad practice and those libraries should be fixed, but if you must use one of them, there are some options available for you to accommodate their limitations:
```
Comment (ide) [inline docs/pages/guides/configuring-statusbar.md:114]: I think we can just say "The status bar is `translucent` by default for Expo projects on Android."

File: docs/pages/versions/v37.0.0/sdk/status-bar.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+- [`StatusBarProps`](#statusbarprops)
+- [`StatusBarStyle`](#statusbarstyle)
+
+## Components
+
+### `StatusBar`
+
+A component that allows you to configure your status bar without directly calling imperative methods like `setBarStyle`.
+
+You will likely have multiple `StatusBar` components mounted in the same app at the same time. For example, if you have multiple screens in your app, you may end up using one per screen. The props of each `StatusBar` component will be merged in the order that they were mounted. This component is built on top of the [StatusBar](https://reactnative.dev/docs/statusbar) component exported from React Native, and it provides defaults that work better for Expo users.
```
Comment (ide) [inline docs/pages/versions/v37.0.0/sdk/status-bar.md:80]: "Expo users" -> "Expo developers" (though also improves the UX for end users of Expo apps)

File: docs/pages/versions/unversioned/sdk/status-bar.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+- [`StatusBarProps`](#statusbarprops)
+- [`StatusBarStyle`](#statusbarstyle)
+
+## Components
+
+### `StatusBar`
+
+A component that allows you to configure your status bar without directly calling imperative methods like `setBarStyle`.
+
+You will likely have multiple `StatusBar` components mounted in the same app at the same time. For example, if you have multiple screens in your app, you may end up using one per screen. The props of each `StatusBar` component will be merged in the order that they were mounted. This component is built on top of the [StatusBar](https://reactnative.dev/docs/statusbar) component exported from React Native, and it provides defaults that work better for Expo users.
```
Comment (ide) [inline docs/pages/versions/unversioned/sdk/status-bar.md:80]: Expo users -> Expo developers

**Review by ide — COMMENTED** [review]
Left some post-commit review comments, mostly on the docs.

**Pull request merged**